website/cue/reference/remap/functions/find_enrichment_table_records.cue

```diff
@@ -0,0 +1,79 @@
+package metadata
+
+remap: functions: find_enrichment_table_records: {
+	category:    "Enrichment"
+	description: """
+		Searches an [enrichment table](\(urls.enrichment_tables_concept)) for rows that match the
+		provided condition.
+
+		\(remap._enrichment_table_explainer)
+		"""
+
+	arguments: [
+		{
+			name:        "table"
+			description: "The [enrichment table](\(urls.enrichment_tables_concept)) to search."
+			required:    true
+			type: ["string"]
+		},
+		{
+			name: "condition"
+			description: """
+				The condition to search on. Since the condition is used at boot time to create
+				indices into the data, these conditions must be statically defined.
+				"""
+			required: true
+			type: ["object"]
+		},
+		{
+			name: "select"
+			description: """
+				A subset of fields from the enrichment table to return. If not specified,
+				all fields are returned.
+				"""
+			required: false
+			type: ["array"]
+		},
+		{
+			name:        "case_sensitive"
+			description: "Whether text fields need to match cases exactly."
+			required:    false
+			type: ["boolean"]
+			default: true
+		},
```

Comment:
This is out-of-scope in terms of code changes, but shouldn't this flag be configurable _per field_ in a search?

That is, I'd expect to write:

```coffee
conditions = {
	"my_field_1": { "value": "my_case_sensitive_value", "case_sensitive": true },
	"my_field_2": { "value": "my_case_insensitive_value", "case_sensitive": false },
}

find_enrichment_table_records("my_table", conditions)
```

As opposed to:

```coffee
conditions = {
	"my_field_1": "my_case_sensitive_value",
	"my_field_2": "my_case_insensitive_value",
}

find_enrichment_table_records("my_table", conditions, case_sensitive: true)
```

The former also allows us to add more options for each condition in the future (e.g. we could add a `match_type` field and have it be either `exact` or `partial` or whatever).

I know this is a very late call for such a change, but this came to mind while reading the Cue docs, and I felt it would be better not to let this opportunity go to waste before we officially release enrichment tables.

Reply:
I think that it is highly unlikely that you would want case sensitive in one field and insensitive in another, and so decided that the extra complexity wasn't worth it.

If a use case comes along we could add the ability to override the global `case_sensitive` on a field basis. I think it is worth holding off until then.

Reply:
While I agree with Jean that users may want to search insensitive based on certain fields, this also seems like something we could do in additive fashion later while preserving compatibility so I'm inclined to release as-is to start collecting user feedback. We can schedule more work around this in the future.

Reply:
Hmm, yeah, we might be able to do this without breaking backward compatibility (or accept a breakage if needed). 👍